pkg/apis/edgeapi.cattle.io/v1alpha1/device_template_revision_types.go

```diff
@@ -0,0 +1,39 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type DeviceTemplateRevision struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   DeviceTemplateRevisionSpec   `json:"spec,omitempty"`
+	Status DeviceTemplateRevisionStatus `json:"status,omitempty"`
+}
+
+type DeviceInfo struct {
+	DeviceKind     string `json:"deviceKind,omitempty"`
+	DeviceVersion  string `json:"deviceVersion,omitempty"`
+	DeviceGroup    string `json:"deviceGroup,omitempty"`
+	DeviceResource string `json:"deviceResource,omitempty"`
+}
```

Comment:
> since those parameters are duplicate with the device template, prefer to move it to labels and we would also be able to filter by its device metadata.

Since the metadata name has a fixed format restriction, it may not be flexible enough to meet the user's naming conventions. like  name  `v1-20200610` is not  allowed